sql/src/main/java/io/crate/expression/reference/sys/cluster/ClusterLicenseExpression.java

```diff
@@ -59,15 +57,12 @@ public NestableInput getChild(String name) {
 
     private void fillChildImplementations(LicenseData licenseData) {
         if (licenseData != null) {
-            childImplementations.put(EXPIRY_DATE, forFunction(ignored -> {
-                if (licenseData.expiryDateInMs() == Long.MAX_VALUE) {
-                    return null;
-                } else {
-                    return licenseData.expiryDateInMs();
-                }
-            }));
-            childImplementations.put(ISSUED_TO, forFunction(ignored -> licenseData.issuedTo()));
-            childImplementations.put(MAX_NODES, forFunction(ignored -> licenseData.maxNumberOfNodes()));
+            var expiryDateInMs = licenseData.expiryDateInMs() == Long.MAX_VALUE
+                ? null
+                : licenseData.expiryDateInMs();
+            childImplementations.put(EXPIRY_DATE, constant(expiryDateInMs));
+            childImplementations.put(ISSUED_TO, constant(licenseData.issuedTo()));
+            childImplementations.put(MAX_NODES, constant(licenseData.maxNumberOfNodes()));
```

Comment:
How does switching from `forFunction` to `constant` fix the issue? Is `LicenseData` mutated?

Reply:
The setNextRow is never called for `FuncExpression` so calling value on child expression will always result in null. I've checked other implementations of `NestedObjectExpression` they never use `forFunction`, so I decided to follow this approach.